ggml-quants.c

```diff
@@ -1273,7 +1273,8 @@ static float make_qkx2_quants(int n, int nmax, const float * restrict x, const f
     float max = x[0];
     float sum_w = weights[0];
     float sum_x = sum_w * x[0];
-    for (int i = 1; i < n; ++i) {
+    // Mark i as volatile to prevent the -O3 optimizer from unrolling this loop and breaking MacOS Sonoma quantization
+    for (volatile int i = 1; i < n; ++i) {
```

Comment:
This works for me on my M2 Macbook Pro:
```suggestion
+#if defined(__APPLE__) && defined(__clang_major__) && __clang_major__ >= 15
+    // use 'volatile' to prevent unroll and work around bugs in Apple clang 15.x.x
+    for (volatile int i = 1; i < n; ++i) {
+#else
     for (int i = 1; i < n; ++i) {
+#endif
```